pyk/src/pyk/kcfg.py

```diff
@@ -253,8 +257,37 @@ def resolve(node_id: str) -> str:
         for verified_ids in dct.get('verified') or []:
             cfg.add_verified(resolve(verified_ids['source']), resolve(verified_ids['target']))
 
+        for alias, id in (dct.get('aliases') or {}).items():
+            cfg.add_alias(name=alias, id=id)
+
         return cfg
 
+    def short_id(self, node: Node) -> str:
+        for alias, hash in self._aliases.items():
+            if node.id == hash:
+                return alias
+        return shorten_hash(node.id)
```

Comment:
This just returns the first alias it finds? What if the user defines an alias, and we have a short-id for the node too? We should probably allow it to keep shortening the aliases?

Also, what should it do if there are two aliases for the same node?

Reply:
@ehildenb 

> What if the user defines an alias, and we have a short-id for the node too? We should probably allow it to keep shortening the aliases?

A short-id is an alias if it has one, otherwise it is the shorten hash. Do you mean we should look for a shorter alias?

> Also, what should it do if there are two aliases for the same node?

I'm not sure. I was thinking we could just disallow that for now.

Reply:
What do we want to use aliases for?

If they are just for pretty printing, then we should store them as `node_id -> alias`, which ensures fast lookup and that there is at most one alias per node.

But probably we also want to use them like `cfg.node(alias)`. In this case it makes sense to admit multiple aliases, so we also need `alias -> node_id`. The other dict used for pretty printing can then be used to store the default alias, with methods that maintain it in a sensible way.

Reply:
Aliases may also passed to any command that takes an id, so I think the second is better.

I'm going to allow multiple aliases, and display all aliases when pretty printing.

Reply:
Regarding maintaining a default alias, I'm not sure its worth the engineering effort. We would need an interface for setting the default alias as well.

Reply:
> I'm going to allow multiple aliases, and display all aliases when pretty printing.

In this case a default alias is not necessary.